gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -163,100 +287,216 @@ cmd VkResult vkCreateRenderPass(
-      description.ColorAttachments[j] = colorAttachments[j]
+      attachment := colorAttachments[j]
+      description.ColorAttachments[j] = AttachmentReference(
+        Attachment: attachment.attachment,
+        Layout: attachment.layout,
+        AspectMask: attachment.aspectMask
+      )
     }
+
     if subpass.pResolveAttachments != null {
       resolveAttachments := subpass.pResolveAttachments[0:subpass.colorAttachmentCount]
       for j in (0 .. subpass.colorAttachmentCount) {
-        description.ResolveAttachments[j] = resolveAttachments[j]
+        attachment := resolveAttachments[j]
+        description.ResolveAttachments[j] = AttachmentReference(
+          Attachment: attachment.attachment,
+          Layout: attachment.layout,
+          AspectMask: attachment.aspectMask
+        )
       }
     }
+
     if (subpass.pDepthStencilAttachment != null) {
-      depth_attachment := subpass.pDepthStencilAttachment[0]
-      description.DepthStencilAttachment = new!VkAttachmentReference(
-        Attachment: depth_attachment.Attachment,
-        Layout:     depth_attachment.Layout)
+      attachment := subpass.pDepthStencilAttachment[0]
+      description.DepthStencilAttachment = new!AttachmentReference(
+        Attachment: attachment.attachment,
+        Layout: attachment.layout,
+        AspectMask: attachment.aspectMask,
+      )
     }
+
     preserveAttachments := subpass.pPreserveAttachments[0:subpass.preserveAttachmentCount]
     for j in (0 .. subpass.preserveAttachmentCount) {
       description.PreserveAttachments[j] = preserveAttachments[j]
     }
+
     renderPass.SubpassDescriptions[i] = description
   }
+
   dependencies := info.pDependencies[0:info.dependencyCount]
   for i in (0 .. info.dependencyCount) {
-    renderPass.SubpassDependencies[i] = dependencies[i]
+    dependency := dependencies[i]
+    renderPass.SubpassDependencies[i] = SubpassDependency(
+      SrcSubpass: dependency.srcSubpass,
+      DstSubpass: dependency.dstSubpass,
+      SrcStageMask: dependency.srcStageMask,
+      DstStageMask: dependency.dstStageMask,
+      SrcAccessMask: dependency.srcAccessMask,
+      DstAccessMask: dependency.dstAccessMask,
+      DependencyFlags: dependency.dependencyFlags,
+      ViewOffset: dependency.viewOffset,
+    )
   }
-  handle := ?
-  if pRenderPass == null { vkErrorNullPointer("VkRenderPass") }
-  pRenderPass[0] = handle
-  renderPass.VulkanHandle = pRenderPass[0]
+
+  if info.pCorrelatedViewMasks != null {
+    pCorrelatedViewMasks := info.pCorrelatedViewMasks[0:info.correlatedViewMaskCount]
+    for i in (0 .. info.correlatedViewMaskCount) {
+      renderPass.CorrelatedViewMasks[i] = pCorrelatedViewMasks[i]
+    }
+  }
+
+  if info.pNext != null {
+    // We do not support any extension for this functionality.
+    next := MutableVoidPtr(as!void*(info.pNext))
+    sType := as!const VkStructureType*(next.Ptr)[0]
+    vkErrorUnrecognizedExtensionPNext(sType)
+  }
```

Comment:
if you look at the generated code in the `mutate.go` You can see that the existing code has exactly the same behavior.

I have actually thought about having a loop to check it but with the way the Vulkan release works, it's very unlikely that we will get two new extensions that are unknown and used at the same time. Therefore I wanted to keep it consistently incomplete error handling.

But if you feel strong about it, I can change. In that case these pNext checks will be the only ones that are able to do it and it may confuse us in the future about which missing pNext are reported all and which are repored with only the first one.

For your suggestion, to be fair, I don't know if that will be compiled and run as we expect. My dealing with gapil is I look at how everything else done and follow them. If you want I can try to make it one line but I don't know if there is any benefit of risking that we get wrong behavior.